module/Activity/src/Activity/Controller/ActivityController.php

```diff
@@ -167,6 +177,44 @@ public function signupAction()
         $this->redirectActivityRequest($id, true, $message);
     }
 
+    public function externalSignupAction()
+    {
+        $id = (int) $this->params('id');
+        $activityService = $this->getServiceLocator()->get('activity_service_activity');
+        $queryService = $this->getServiceLocator()->get('activity_service_activityQuery');
+        $signupService = $this->getServiceLocator()->get('activity_service_signup');
+
+        $activity = $queryService->getActivity($id);
+
+        $translator = $activityService->getTranslator();
+
+        //Assure the form is used
+        if (!$this->getRequest()->isPost()) {
+            $error = $translator->translate('Use the form to subscribe');
+            $this->redirectActivityRequest($id, false, $error);
+            return;
+        }
+
+        $form = $signupService->getExternalForm($activity->getFields());
+        $form->setData($this->getRequest()->getPost());
+
+        //Assure the form is valid
+        if (!$form->isValid()) {
+            $error = $translator->translate('Invalid form');
```

Comment:
Just showing only this error message isn't very helpful, it would help if additionally the forum errors are shown in the view.